src/plugins/data/public/search/expressions/esdsl.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  KibanaContext,
+  ExpressionFunctionDefinition,
+} from '../../../../../plugins/expressions/public';
+
+import { getSearchService, getUiSettings } from '../../services';
+import { EsRawResponse } from './es_raw_response';
+import { RequestStatistics, RequestAdapter } from '../../../../inspector/common';
+import { IEsSearchResponse } from '../../../common/search/es_search';
+import { buildEsQuery, getEsQueryConfig } from '../../../common/es_query/es_query';
+
+const name = 'esdsl';
+
+type Input = KibanaContext | null;
+type Output = Promise<EsRawResponse>;
+
+interface Arguments {
+  dsl: string;
+}
+
+export type EsdslExpressionFunctionDefinition = ExpressionFunctionDefinition<
+  typeof name,
+  Input,
+  Arguments,
+  Output
+>;
+
+export const esdsl = (): EsdslExpressionFunctionDefinition => ({
+  name,
+  type: 'es_raw_response',
+  inputTypes: ['kibana_context', 'null'],
+  help: i18n.translate('data.search.esdsl.help', {
+    defaultMessage: 'Run Elasticsearch request',
+  }),
+  args: {
+    dsl: {
+      types: ['string'],
+      aliases: ['_', 'q', 'query'],
+      help: i18n.translate('data.search.esdsl.q.help', {
+        defaultMessage: 'Query DSL',
+      }),
+      required: true,
+    },
+  },
+  async fn(input, args, { inspectorAdapters, abortSignal }) {
+    const searchService = getSearchService();
+
+    const dsl = JSON.parse(args.dsl);
+    if (!dsl.body) dsl.body = {};
+
+    const dslBody = dsl.body;
```

Comment:
The argument that you're calling `body` should be the only argument. Everything else should be top-level arguments to the expression function, such as:

`esdsl index="my_index" dsl='{ "query": { "match_all": {} }, "size": 0, "aggs": { } }'`